Base+Meta: Generate emoji.txt at build time

squeek502 opened the pull request:
Instead of manually updating emoji.txt whenever new emoji are added,
we use Unicode's [`emoji-test.txt`](https://www.unicode.org/Public/emoji/14.0/emoji-test.txt) to generate `emoji.txt` on each build,
including only the emojis that Serenity supports at that time.

By using `emoji-test.txt`, we can also include all forms of each emoji
(fully-qualified, minimally-qualified, and unqualified) which can be
helpful when double-checking how certain forms are handled.

Serenity-specific emoji not included in Unicode's `emoji-test.txt` were moved to `emoji-serenity.txt` which can continue to be manually maintained.

`emoji.txt` itself has been removed and added to `.gitignore`.

Example portion of the generated `emoji.txt`:

```
# group: Smileys & Emotion

# subgroup: face-smiling
😀 - U+1F600 GRINNING FACE (fully-qualified)
😃 - U+1F603 GRINNING FACE WITH BIG EYES (fully-qualified)
😄 - U+1F604 GRINNING FACE WITH SMILING EYES (fully-qualified)
😁 - U+1F601 BEAMING FACE WITH SMILING EYES (fully-qualified)
😆 - U+1F606 GRINNING SQUINTING FACE (fully-qualified)
😅 - U+1F605 GRINNING FACE WITH SWEAT (fully-qualified)
🤣 - U+1F923 ROLLING ON THE FLOOR LAUGHING (fully-qualified)
😂 - U+1F602 FACE WITH TEARS OF JOY (fully-qualified)
🙂 - U+1F642 SLIGHTLY SMILING FACE (fully-qualified)
🙃 - U+1F643 UPSIDE-DOWN FACE (fully-qualified)
🫠 - U+1FAE0 MELTING FACE (fully-qualified)
😉 - U+1F609 WINKING FACE (fully-qualified)
😊 - U+1F60A SMILING FACE WITH SMILING EYES (fully-qualified)
😇 - U+1F607 SMILING FACE WITH HALO (fully-qualified)

# subgroup: face-affection
🥰 - U+1F970 SMILING FACE WITH HEARTS (fully-qualified)
😍 - U+1F60D SMILING FACE WITH HEART-EYES (fully-qualified)
🤩 - U+1F929 STAR-STRUCK (fully-qualified)
😘 - U+1F618 FACE BLOWING A KISS (fully-qualified)
😗 - U+1F617 KISSING FACE (fully-qualified)
😚 - U+1F61A KISSING FACE WITH CLOSED EYES (fully-qualified)
😙 - U+1F619 KISSING FACE WITH SMILING EYES (fully-qualified)
🥲 - U+1F972 SMILING FACE WITH TEAR (fully-qualified)
```

linusg commented:
Very cool, but please don't commit generated content - let's remove it from the repo instead :)

squeek502 commented:
> Very cool, but please don't commit generated content - let's remove it from the repo instead :)

I'm fine with either way, but it might be worth noting that there seems to be some precedence for keeping it committed as well, i.e. some man pages are generated and kept in the repo via [generate_manpages.sh](https://github.com/SerenityOS/serenity/blob/master/Base/root/generate_manpages.sh) and [export-argsparser-manpages.sh](https://github.com/SerenityOS/serenity/blob/master/Meta/export-argsparser-manpages.sh).

linusg commented:
That's different, they're generated _as a base_ for being extended - no-one wants to write out all options by hand. This is not the case here.

squeek502 commented:
Updated:
- Deleted emoji.txt and added it to `.gitignore`
- Improved the CMake side of things to only run the `emoji.txt` generation when necessary (i.e. `emoji.txt` doesn't exist or a file was added or removed from `Base/res/emoji`).
